api/s2n.h

```diff
@@ -421,6 +421,97 @@ extern int s2n_connection_set_client_auth_type(struct s2n_connection *conn, s2n_
 S2N_API
 extern int s2n_connection_get_client_cert_chain(struct s2n_connection *conn, uint8_t **der_cert_chain_out, uint32_t *cert_chain_len);
 
+/**
+ * Sets the initial number of session tickets to send after a handshake.
+ *
+ * @param config A pointer to the config object.
+ * @param num The number of session tickets that will be sent.
+ */
+S2N_API
+extern int s2n_config_set_initial_ticket_count(struct s2n_config *config, uint8_t num);
+
+/**
+ * Increases the amount of session tickets to send after a handshake.
+ *
+ * @param conn A pointer to the s2n_connection object.
+ * @param num The number of additional session tickets to send.
+ */
+S2N_API
+extern int s2n_connection_add_new_tickets_to_send(struct s2n_connection *conn, uint8_t num);
+
+/**
+ * Sets the keying material lifetime for session tickets so that one session doesn't get re-used ad infinitum.
+ * The default value is one week.
+ *
+ * @param conn A pointer to the s2n_connection object.
+ * @param lifetime_in_secs Lifetime of keying material in seconds.
+ */
+S2N_API
+extern int s2n_connection_set_server_keying_material_lifetime(struct s2n_connection *conn, uint32_t lifetime_in_secs);
+
+struct s2n_session_ticket;
+
+/**
+ * Callback function for receiving a session ticket
+ *
+ * # Safety
+ *
+ * `ctx` MUST be cast into the same type of pointer that was originally created.
+ * `ticket` is an internal pointer that can only be used until this function exits.
+ *
```

Comment:
I'm not sure about the phrasing here. Idk if I'd know what you meant if I didn't already know what you mean. Maybe something like
```suggestion
 *
 * `ctx` is a void pointer and the caller is responsible for ensuring it is cast to the correct type
 * `ticket` is valid only within the scope of this callback
 *
```